website/pages/docs/internals/integrated-storage.mdx

```diff
@@ -0,0 +1,189 @@
+using BoltDB is that it allows Vault's snapshots to be very light weight. Since
+Vault's data is already persisted to disk in BoltDB the snapshot process just
+needs to truncate the raft logs.
+
+Consensus is fault-tolerant while a cluster has quorum.
+If a quorum of nodes is unavailable, it is impossible to process log entries or reason
+about peer membership. For example, suppose there are only 2 peers: A and B. The quorum
+size is also 2, meaning both nodes must agree to commit a log entry. If either A or B
+fails, it is now impossible to reach quorum. This means the cluster is unable to add
+or remove a node or to commit any additional log entries. This results in
+*unavailability*. At this point, manual intervention would be required to remove
+either A or B and to restart the remaining node in bootstrap mode.
+
+A Raft cluster of 3 nodes can tolerate a single node failure while a cluster
+of 5 can tolerate 2 node failures. The recommended configuration is to either
+run 3 or 5 Vault servers per cluster. This maximizes availability without
+greatly sacrificing performance. The [deployment table](#deployment_table) below
+summarizes the potential cluster size options and the fault tolerance of each.
+
+In terms of performance, Raft is comparable to Paxos. Assuming stable leadership,
+committing a log entry requires a single round trip to half of the cluster.
+Thus, performance is bound by disk I/O and network latency. 
+
+### Raft in Vault
+
+When getting started, a single Vault server is
+[initialized](/docs/commands/operator/init/#operator-init). At this point the
+cluster is of size 1 which allows the node to self-elect as a leader. Once a
+leader is elected, other servers can be added to the peer set in a way that
+preserves consistency and safety. 
+
+The join process is how new nodes are added to the vault cluster, it uses an
+encrypted challenge/answer workflow. To accomplish this, all nodes in a single
+raft cluster must share the same seal configuration. If using an Auto Unseal the
+join process can use the configured seal to automatically decrypt the challenge
+and respond with the answer. If using a Shamir seal the unseal keys must be
+provided to the node attempting to join the cluster before it can decrypt the
+challenge and respond with the decrypted answer. 
+
+Since all servers participate as part of the peer set, they all know the current
+leader. When an API request arrives at a non-leader server, the request is
+forwarded to the leader.
+
+Similar to other storage backends, data that is written to the Raft log and FSM
+will be encrypted by Vault's barrier.
+
+### Deployment Table
+
+Below is a table that shows quorum size and failure tolerance for various
+cluster sizes. The recommended deployment is either 3 or 5 servers. A single
+server deployment is _**highly**_ discouraged as data loss is inevitable in a
+failure scenario.
+
+<table class="table table-bordered table-striped">
```

Comment:
You can use a markdown table here if you'd like: https://help.github.com/en/github/writing-on-github/organizing-information-with-tables